src/test/java/com/google/alpollo/AnalysisTest.java

```diff
@@ -214,6 +234,85 @@ public void sentimentIsMixed() {
+  public void foundArtistWithIncompleteName() throws IOException {
+    when(request.getParameter(SEARCH_STRING)).thenReturn(INCOMPLETE_ARTIST_NAME);
+    when(request.getParameter(TYPE)).thenReturn(ARTIST);
+    autocompleteServletUnderTest.doPost(request, response);
+    String responseString = responseWriter.toString();
+
+    List<String> actual = gson.fromJson(responseString, new TypeToken<List<String>>(){}.getType());
+    List<String> expected = Arrays.asList("Dr. Dre", "Michael Clarke", "Eminem", "Emine Erdoğan",
+        "Emine Gülbahar Hatun", "Emine Ülker Tarhan", "Mihai Eminescu", "Emine Şenlikoğlu", 
+        "Cardinal Richelieu", "Gönül Ülkü");
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void foundArtistWithWrongName() throws IOException {
+    when(request.getParameter(SEARCH_STRING)).thenReturn(WRONG_ARTIST_NAME);
+    when(request.getParameter(TYPE)).thenReturn(ARTIST);
+    autocompleteServletUnderTest.doPost(request, response);
+    String responseString = responseWriter.toString();
+
+    List<String> actual = gson.fromJson(responseString, new TypeToken<List<String>>(){}.getType());
+    List<String> expected = Arrays.asList();
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void artistIsABand() throws IOException {
+    when(request.getParameter(SEARCH_STRING)).thenReturn(INCOMPLETE_BAND_NAME);
+    when(request.getParameter(TYPE)).thenReturn(ARTIST);
+    autocompleteServletUnderTest.doPost(request, response);
+    String responseString = responseWriter.toString();
+
+    List<String> actual = gson.fromJson(responseString, new TypeToken<List<String>>(){}.getType());
+    List<String> expected = Arrays.asList("Rammstein", "Otthein Rammstedt", "Rammstein", "Rammstedt");
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void searchSongWithIncompleteName() throws IOException {
+    when(request.getParameter(SEARCH_STRING)).thenReturn(INCOMPLETE_SONG_NAME);
+    when(request.getParameter(TYPE)).thenReturn(SONG);
+    autocompleteServletUnderTest.doPost(request, response);
+    String responseString = responseWriter.toString();
+
+    List<String> actual = gson.fromJson(responseString, new TypeToken<List<String>>(){}.getType());
+    List<String> expected = Arrays.asList("Master of the House (Kono Uchi no Shu)", "Master Of Tiramisu",
+        "Master of Xone", "Master of No Mercy", "Master of the Dead", "Master of Time", "Master of Destiny",
+        "Master of Sleep", "Master of Art", "Master of the Universe");
+
+    Assert.assertEquals(expected, actual);
+  }  
+
+  @Test
+  public void autocompleteBadTypeRequest() throws IOException {
+    when(request.getParameter(SEARCH_STRING)).thenReturn(INCOMPLETE_SONG_NAME);
+    when(request.getParameter(TYPE)).thenReturn(PAINTING);
+    autocompleteServletUnderTest.doPost(request, response);
+    
+    verify(response).sendError(
+        HttpServletResponse.SC_BAD_REQUEST,
+        "Type not supported.");
+  }
```

Comment:
This test is not very useful as you are the one creating the request, so that should never happen

Reply:
Should I just delete it?

Reply:
Yes, I don't think it brings any value